core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -282,11 +291,85 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
     }
 
   /**
-   * Evaluates the specified block, catching exceptions of the specified type. Uncaught exceptions are propagated.
+   * Evaluates the specified block, catching exceptions of the specified type.
+   *
+   * This method is considered unsafe because uncaught exceptions are propagated
+   * outside of `F`. While the name does not currently reflect this, this may change
+   * at a future date.
    */
   def catchOnly[T >: Null <: Throwable]: CatchOnlyPartiallyApplied[T, F, E] =
     new CatchOnlyPartiallyApplied[T, F, E](this)
 
+  /**
+   * Evaluates the specified block, catching exceptions of the specified type.
+   *
+   * Exceptions of type `T` are raised in `F`, other non-fatal exceptions are raised in `G`
+   *
+   * Note: fatal exceptions (as defined by `scala.util.control.NonFatal`) will be propagated
+   */
+  def catchOnlySafe[G[_], T >: Null <: Throwable]: CatchOnlySafePartiallyApplied[T, G, F, E] =
```

Comment:
Personally, I find such a naming quite confusing. I feel in Cats everything is supposed to be safe, _unless_ stated and marked otherwise.

Reply:
No argument, I just needed to name it something so I could put a safe variant of `catchOnly` in there so we can figure out if it's something worth adding or not, as renaming the existing method to `catchOnlyUnsafe` would break bincompat